torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp

```diff
@@ -17,171 +18,881 @@ namespace fuser {
+          // reversible operations e.g.:
+          //    [id{3} id{5} id{6}] -> merge(1, 0)
+          // -> [id{5*3} id{6}] -> split(0, 5)
+          // -> [id{5} id{3} id{6}] -> transpose(0, 1)
+          // -> [id{3} id{5} id{6}]
+          // However we don't try and capture cases like this correcly, we'd
+          // just reduce this down to only the iter domain of size 6 mapping.
+          //
+          // Clear outer and left of outer
+          clear_left_of(find_outer_it + 1);
+          continue;
+        }
+      }
+
+      // Find the position inner would have to have to be considered ordered
+      // relative to outer
+      auto pos_after_outer = outer_pos + 1;
+      for (; pos_after_outer < ids.size(); pos_after_outer++) {
+        if (ids[pos_after_outer]->isBroadcast()) {
+          // Skip broadcast axes as they must not have been concretized in the
+          // reference. We remove dimensions that underwent a concretization as
+          // well as the dimensions to the left of that.
+          continue;
+        }
+        break;
+      }
+
+      if (inner_pos != pos_after_outer) {
+        // Nothing to the left of inner could be continuous.
+        //
+        // Clear left of inner
+        clear_left_of(find_inner_it);
+      }
+
+      if (hasPartialExtent(split->inner())) {
+        // Nothing to the left of inner can map, clear left of inner
+        clear_left_of(find_inner_it);
+      }
+
+      if (find_outer_it != ids.end() && find_inner_it != ids.end()) {
+        // Both dimensions map, inner dimension maps fully. For more context see
+        // the comment:
+        //  projectIdToRFactor
+        //    if (find_outer_it != ids.end() && find_inner_it != ids.end() &&
+        //      !hasPartialExtent(merge->inner())) {
+        //      Comment here.
+
+        // Removed outer, so inner shifts left
+        ids[inner_pos] = split->in();
+        ids.erase(find_outer_it);
+        auto in_pos = inner_pos - 1;
+        if (hasPartialExtent(split->outer())) {
+          // Outer dimension has a partial map, but inner dimension is full,
+          // split->in should be partially mapped.
+          partial_projected_extent_[split->in()] =
+              SimplifyingIrBuilder::mulExpr(
+                  getMaybePartialMappedExtent(split->outer()),
+                  split->inner()->extent());
+          // Clear to the left of split in since it partially maps
+          if (inner_pos > 0) {
+            ids.erase(ids.begin(), ids.begin() + in_pos);
+          }
+        }
+      } else {
+        // Only inner matches, mark a partial match
+        ids[inner_pos] = split->in();
+        // Mark the partial match, inner could already be a partial match
+        partial_projected_extent_[split->in()] =
+            getMaybePartialMappedExtent(split->inner());
+        // Clear to the left of split in since it partially maps
+        ids.erase(ids.begin(), ids.begin() + inner_pos);
```

Comment:
I think inner_pos should already be 0 at this point? We can try the following:
```suggestion
        TORCH_INTERNAL_ASSERT(inner_pos == 0);
```

Reply:
Yeah, that's my understanding.

Reply:
Agreed, but I'm officially scared of modifying this code because of a redundant line. Commented.